src/osemgrep/cli/CLI.ml

```diff
@@ -190,8 +190,7 @@ let dispatch_subcommand argv =
         | "login" when experimental -> Login_subcommand.main subcmd_argv
         | "logout" when experimental -> Logout_subcommand.main subcmd_argv
         | "publish" when experimental -> missing_subcommand ()
-        (* TODO: next target for not requiring the 'when experimental' guard!*)
-        | "lsp" when experimental -> Lsp_subcommand.main subcmd_argv
```

Comment:
can we delete lsp.py then?

Reply:
There is still metrics code in there. So technically yes, but I'd prefer to wait till we port that too

Reply:
osemgrep is already sending metrics data. See CLI.ml
maybe you just need to enable it by default for lsp in Lsp_subcommand.run.
for the scan it's enabled via   Metrics_.configure conf.metrics;
 but only if you did --metrics=on, or if you use the registry.

Reply:
And technically now that this PR is merged, lsp.py is dead code; It's just not executed anymore unless you explicitely used --legacy.